dotnet/src/Functions/Functions.OpenApi.Extensions/Extensions/CopilotAgentPluginKernelExtensions.cs

```diff
@@ -159,13 +160,47 @@ await DocumentLoader.LoadDocumentFromUriAsStreamAsync(parsedDescriptionUrl,
 #pragma warning disable CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
             var operationRunnerHttpClient = HttpClientProvider.GetHttpClient(openApiFunctionExecutionParameters?.HttpClient ?? kernel.Services.GetService<HttpClient>());
 #pragma warning restore CA2000
+            static IDictionary<string, string>? CopilotAgentPluginHeadersFactory(RestApiOperation operation, IDictionary<string, object?> arguments, RestApiOperationRunOptions? options)
+            {
+                var graphAllowedHosts = new[]
+                {
+                    "graph.microsoft.com",
+                    "graph.microsoft.us",
+                    "dod-graph.microsoft.us",
+                    "graph.microsoft.de",
+                    "microsoftgraph.chinacloudapi.cn",
+                    "canary.graph.microsoft.com",
+                    "graph.microsoft-ppe.com"
+                };
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
+                var isGraphHost = graphAllowedHosts.Contains(options.ApiHostUrl.Host);
+#pragma warning restore CS8602 // Dereference of a possibly null reference.
+                if (!isGraphHost)
+                {
+                    return null;
+                }
```

Comment:
its better to deal with the possible nullable reference rather than suppress it.
```suggestion
                if (string.IsNullOrEmpty(options?.ApiHostUrl?.Host) || graphAllowedHosts.Contains(options.ApiHostUrl.Host))
                {
                    return null;
                }
```